docs/dev/connectors/pubsub.md

```diff
@@ -0,0 +1,110 @@
+specific language governing permissions and limitations
+under the License.
+-->
+
+This connector provides a Source and Sink that can read from and write to
+[Google PubSub](https://cloud.google.com/pubsub). To use this connector, add the
+following dependency to your project:
+
+{% highlight xml %}
+<dependency>
+  <groupId>org.apache.flink</groupId>
+  <artifactId>flink-connector-pubsub{{ site.scala_version_suffix }}</artifactId>
+  <version>{{ site.version }}</version>
+</dependency>
+{% endhighlight %}
+
+Note that the streaming connectors are currently not part of the binary
+distribution. See
+[here]({{ site.baseurl }}/dev/linking.html)
+for information about how to package the program with the libraries for
+cluster execution.
+
+#### PubSub Source
+
+The connector provides a Source for reading data from Google PubSub to Apache Flink.
+PubSub has an Atleast-Once guarantee and as such this is the best we can do
+right now without adding a lot of extra complexity.
+
+The class `PubSubSource(…)` has a builder to create PubSubsources. `PubSubSource.newBuilder()`
+
+There are several optional methods to alter how the PubSubSource is created,
+the bare minimum is to provide a google project, pubsub subscription and a way to deserialize the PubSubMessages.
+
+Example:
+
+<div class="codetabs" markdown="1">
+<div data-lang="java" markdown="1">
+{% highlight java %}
+StreamExecutionEnvironment streamExecEnv = StreamExecutionEnvironment.getExecutionEnvironment();
+
+DeserializationSchema<SomeObject> deserializationSchema = (...);
+SourceFunction<SomeObject> pubsubSource = PubSubSource.<SomeObject>newBuilder()
+                                                      .withDeserializationSchema(deserializationSchema)
+                                                      .withProjectSubscriptionName("google-project-name", "pubsub-subscription")
+                                                      .build();
+
+streamExecEnv.addSource(pubsubSource);
+{% endhighlight %}
+</div>
+</div>
+
+#### PubSub Sink
+
+The connector provides a Sink for writing data to PubSub.
+
+The class `PubSubSource(…)` has a builder to create PubSubsources. `PubSubSource.newBuilder()`
+
+This builder works in a similar way to the PubSubSource.
+Example:
+
+<div class="codetabs" markdown="1">
+<div data-lang="java" markdown="1">
+{% highlight java %}
+StreamExecutionEnvironment streamExecEnv = StreamExecutionEnvironment.getExecutionEnvironment();
+
+SerializationSchema<SomeObject> serializationSchema = (...);
+SinkFunction<SomeObject> pubsubSink = PubSubSink.<SomeObject>newBuilder()
+                                                  .withSerializationSchema(serializationSchema)
+                                                  .withTopicName("pubsub-topic-name")
+                                                  .withProjectName("google-project-name")
+                                                  .build()
+
+streamExecEnv.addSink(pubsubSink);
```

Comment:
The `StreamExecutionEnvironment` doesn't have a method `addSink()`.

Reply:
Fixed